test/e2e/utils/helpers.go

```diff
@@ -436,41 +436,139 @@ func GetScyllaConfigClient(ctx context.Context, client corev1client.CoreV1Interf
 	return configClient, nil
 }
 
-func GetHostsAndUUIDs(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) ([]string, []string, error) {
+func GetBroadcastAddresses(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) ([]string, error) {
 	serviceList, err := client.Services(sc.Namespace).List(ctx, metav1.ListOptions{
 		LabelSelector: GetMemberServiceSelector(sc.Name).String(),
 	})
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	var hosts []string
-	var uuids []string
-	for _, s := range serviceList.Items {
-		host := s.Spec.ClusterIP
+	var broadcastAddresses []string
+	for _, svc := range serviceList.Items {
+		broadcastAddress, err := GetBroadcastAddress(ctx, client, sc, &svc)
+		if err != nil {
+			return nil, fmt.Errorf("can't get broadcast address of Service %q: %w", naming.ObjRef(&svc), err)
+		}
 
-		if host == corev1.ClusterIPNone {
-			pod, err := client.Pods(sc.Namespace).Get(ctx, s.Name, metav1.GetOptions{})
-			if err != nil {
-				return nil, nil, fmt.Errorf("can't get pod %q: %w", naming.ManualRef(sc.Namespace, s.Name), err)
-			}
-			host = pod.Status.PodIP
+		broadcastAddresses = append(broadcastAddresses, broadcastAddress)
+	}
+
+	return broadcastAddresses, nil
+}
+
+func GetBroadcastAddress(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster, svc *corev1.Service) (string, error) {
+	host := svc.Spec.ClusterIP
+
+	if host == corev1.ClusterIPNone {
+		pod, err := client.Pods(sc.Namespace).Get(ctx, svc.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", fmt.Errorf("can't get pod %q: %w", naming.ManualRef(sc.Namespace, svc.Name), err)
 		}
+		host = pod.Status.PodIP
+	}
+
+	configClient, err := GetScyllaConfigClient(ctx, client, sc, host)
+	if err != nil {
+		return "", fmt.Errorf("can't create scylla config client with host %q: %w", host, err)
+	}
+	broadcastAddress, err := configClient.BroadcastAddress(ctx)
+	if err != nil {
+		return "", fmt.Errorf("can't get broadcast_address of host %q: %w", host, err)
+	}
+
+	return broadcastAddress, nil
+}
+
+func GetUUIDs(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) ([]string, error) {
+	_, uuids, err := GetBroadcastRPCAddressesAndUUIDs(ctx, client, sc)
+
+	return uuids, err
+}
```

Comment:
I'd personally drop this as you can always ignore the return var on the caller side and it takes the same args but I don't feels strong

Reply:
done